resources/repo.rb

```diff
@@ -0,0 +1,77 @@
+property :repo_name, String, name_attribute: true
+property :repo_type, String #, default = 'maven-hosted' # FIXME
+property :config, Hash
+property :api_url # TODO
+property :api_user
+property :api_password
+
+def apiclient
+  @apiclient ||= Nexus3::Api.new(api_url, api_user, api_password)
+end
+
+load_current_value do |new_resource|
+  nexus_api 'list_repo_script' do
+    action :run
+  end
+end
+
+action :create do
+  # This will return a Null object if the repository does not exist (so empty JSON?)
+  list_repo_script = <<EOS
+    import groovy,json.JsonOutput
+    JsonOutput.toJson(repository.repositoryManager.get(#{new_resource.repo_name}))
+EOS
+
+  create_or_update_repo_script = <<EOS
+    if (repository.repositoryManager.get(#{new_resource.repo_name}) == nil) {
+        # create
+        def config = new Configuration(
+            repositoryName: #{new_resource.repo_name},
+            recipeName: "FIXME"
+        )
+        repository.createHosted(config)
```

Comment:
What about repository? Maybe you should call the more generic `createRepository` method instead